File: lab/python/matematica_e_operadores/maior_valor_lido.py
def maior_valor_lido(qtd):
    maior = float(input("Digite um valor: "))
    for _ in range(qtd - 1):
        valor = float(input("Digite um valor: "))
        if valor > maior:
            maior = valor
    return maior


def maior_valor_lido_comprimido(qtd):
    maior = float(input("Digite um valor: "))
    for _ in range(qtd - 1):
        valor = float(input("Digite um valor: "))
        maior = valor if valor > maior else maior
    return maior


def maior_valor_lido_lista(qtd):
    valores = []
    for _ in range(qtd):
        valor = float(input("Digite um valor: "))
        valores.append(valor)

    maior = valores[0]
    for valor in valores:
        if valor > maior:
            maior = valor

    return maior


print("Maior =", maior_valor_lido(5))
print("Maior =", maior_valor_lido_comprimido(5))
print("Maior =", maior_valor_lido_lista(5))

File: lab/javascript/matematica_e_operadores/maior_valor_lido.js
const prompt = require("prompt-sync")({ sigint: true });

// Versão 1 — normal
function maior_valor_lido(qtd) {
  let maior = parseFloat(prompt("Digite um valor: "));
  for (let i = 0; i < qtd - 1; i++) {
    let valor = parseFloat(prompt("Digite um valor: "));
    if (valor > maior) {
      maior = valor;
    }
  }
  return maior;
}

// Versão 2 — comprimida
function maior_valor_lido_comprimido(qtd) {
  let maior = parseFloat(prompt("Digite um valor: "));
  for (let i = 0; i < qtd - 1; i++) {
    let valor = parseFloat(prompt("Digite um valor: "));
    maior = valor > maior ? valor : maior;
  }
  return maior;
}

// Versão 3 — usando lista (array)
function maior_valor_lido_lista(qtd) {
  const valores = [];
  for (let i = 0; i < qtd; i++) {
    let valor = parseFloat(prompt("Digite um valor: "));
    valores.push(valor);
  }

  let maior = valores[0];
  for (let valor of valores) {
    if (valor > maior) {
      maior = valor;
    }
  }

  return maior;
}

console.log("Maior =", maior_valor_lido(5));
console.log("Maior =", maior_valor_lido_comprimido(5));
console.log("Maior =", maior_valor_lido_lista(5));
